packages/malloy/src/lang/parse-malloy.ts

```diff
@@ -61,13 +62,16 @@ class ParseErrorHandler implements ANTLRErrorListener<Token> {
     msg: string,
     _e: unknown
   ) {
+    const range = offendingSymbol
+      ? rangeFromToken(offendingSymbol)
+      : {
+          start: { line: 0, character: 0 },
+          end: { line: 0, character: 0 },
+        };
     const error: LogMessage = {
-      sourceURL: this.sourceURL,
       message: msg,
-      begin: {
-        line: line,
-        char: charPositionInLine,
-      },
```

Comment:
There was previously no `end` specified on syntax errors, so the IDE (language server diagnostic code, specifically) was filling in an `end` of the end of the line.